platform/admin/ui/src/main/webapp/js/models/FileHelper.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global define,FileReader*/
+define([
+    'underscore',
+    'backbone'
+], function (_, Backbone) {
+
+    var FileHelper = Backbone.Model.extend({
+        isValid: function () {
+            return this.get('name') !== undefined;
+        },
+        resolveType: function (name, type) {
+            if (type === '') {
+                if (name.lastIndexOf('kar') === (name.length - 3)) {
+                    return 'KAR';
+                } else if (name.lastIndexOf('jar') === (name.length - 3)) {
+                    return 'JAR';
+                }
+            }
+            return type;
+        },
+        setData: function (data) {
+            var file = this.file = data.files[0];
+            this.set({
+                name: file.name,
+                size: file.size,
+                type: this.resolveType(file.name.toLowerCase(), file.type)
+            });
+        },
+        load: function (done) {
+            var reader = new FileReader();
+            reader.onloadend = _.bind(function () {
+                this.set('data', reader.result.split(',')[1]);
+                done();
+            }, this);
+            reader.readAsDataURL(this.file);
+        },
+        toJSON: function () {
```

Comment:
Why did we write our own toJSON function? If you need to exclude something from the resulting json, there are options that you can pass to a backbone association model to include only specific attributes.